external/fv3fit/fv3fit/keras/_models/dense.py

```diff
@@ -95,9 +125,16 @@ def train_dense_model(
         unstacked_dims=["z"],
         n_halo=0,
     )
+    # arrays used in build_model should be the full inputs and outputs before clipping
+    X, y = train_data[0]
+
+    # y train data should be clipped when used as target to fit train_model
+    train_data, validation_data = _clip_data_outputs(
+        hyperparameters, train_data, validation_data  # type: ignore
+    )
+
```

Comment:
Using `validation_data[0]` before meant validation data is always in-memory, which makes sense because it's used for every batch. Can you add a cast of validation_data to `tuple` somewhere, which occurs no matter what (unlike train_data which is cast conditionally)?